### PR TITLE
Add expanded boolean attribute to tailwind plugin

### PR DESCRIPTION
**Pull request by neefrehman**
After testing out the new `Disclosure` components, I see that the tailwind plugin does not support styling the `expanded` state, leaving me to style using `data-[expanded]`. This PR adds `expanded` top the list of boolean states.

Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Review by reidbarber — CHANGES_REQUESTED** [review]
Thanks for adding this!

You'll need to [sign the CLA](https://github.com/adobe/react-spectrum/pull/7152/checks?check_run_id=31188193386) and close/re-open this for the build to pass. 

~Also, it would be great if you could add this to the snapshot tests at `packages/tailwindcss-react-aria-components/src/index.test.js`. If you're having issues with that then no worries, we can add it.~ Beat me to it :)

**Comment by neefrehman** [comment]
@reidbarber thanks! Just done those

**Comment by neefrehman** [comment]
Actually @reidbarber I think I'll need some help with the tests after all, if thats ok? I'm not fully understanding the failures

**Comment by reidbarber** [comment]
@neefrehman I'll take a closer look. The tests seem to pass for me locally. It's likely a whitespace issue or something similar.

File: packages/tailwindcss-react-aria-components/src/index.test.js [hunk 1]
```diff
@@ -399,6 +407,10 @@ test('variants with prefix', async () => {
     --tw-bg-opacity: 1;
     background-color: rgb(255 0 0 / var(--tw-bg-opacity))
 }
+.rac-expanded\:bg-red[data-expanded] {
```
Comment (reidbarber) [inline packages/tailwindcss-react-aria-components/src/index.test.js:410]: I think you just need to add a usage of `rac-expanded:bg-red` to line 311 for this one to pass. I tried to push to your branch, but didn't have permission.

**Review by reidbarber — COMMENTED** [review]
Should be a quick fix

File: packages/tailwindcss-react-aria-components/src/index.test.js (hunk 1 above)
Comment (neefrehman) [reply packages/tailwindcss-react-aria-components/src/index.test.js:410]: Thanks @reidbarber! It turns out I also needed to add the variant to `nnativeVariantSelectors` in the source

**Review by reidbarber — APPROVED** [review]
Thanks!!

**Review by snowystinger — APPROVED** [review]

**Pull request merged**